tokio/src/sync/mpsc/unbounded.rs

```diff
@@ -172,6 +172,41 @@ impl<T> UnboundedReceiver<T> {
         poll_fn(|cx| self.poll_recv(cx)).await
     }
 
+    /// Receives the next values for this receiver and extends `buffer`.
+    ///
+    /// This method returns the number of values populated in `buffer`;
+    /// it returns 0 if the channel has been closed and there
+    /// are no remaining messages in the channel.
+    /// If there are no messages in the channel's buffer, but the channel has
+    /// not yet been closed, this method will sleep until a message is sent or
+    /// the channel is closed.
```

Comment:
I would split this into several paragraphs, like how it is done in the documentation for `recv`. In fact, it would be fine if you copy the documentation for `recv` and adjust it.

Reply:
I reworked the documentation based on `recv`, but changed the phrase "channel's buffer" to "channel's queue" to avoid confusion with the parameter named `buffer`.

Reply:
Sounds good!